sda-commons-server-dropwizard/src/main/java/org/sdase/commons/server/dropwizard/bundles/GenericEnvironmentMapper.java

```diff
@@ -0,0 +1,129 @@
+package org.sdase.commons.server.dropwizard.bundles;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import io.dropwizard.configuration.ConfigurationSourceProvider;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.UncheckedIOException;
+import java.lang.reflect.Type;
+import java.util.List;
+import java.util.Map;
+import org.sdase.commons.server.dropwizard.bundles.JacksonTypeScanner.MappableField;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class GenericEnvironmentMapper<T> implements ConfigurationSourceProvider {
+
+  private static final Logger LOG = LoggerFactory.getLogger(GenericEnvironmentMapper.class);
+
+  private final ConfigurationSourceProvider delegate;
+  private final Class<T> configurationClass;
+  // as in io.dropwizard.configuration.YamlConfigurationFactory
+  private final YAMLFactory yamlFactory = new YAMLFactory();
+  private final ObjectMapper mapper =
```

Comment:
I noticed during profiling that it's quite expensive to create new instances of the ObjectMapper. Would it be possible to re-use an existing one?

Reply:
I don't think so. Dropwizard creates their own as well for parsing the config. And that is hidden in the config reader. I think it is intentional to not use the OM that is prepared for the REST API. I don't think it is an issue because it will only happen once for each app start.